server/tests/phase.service.test.js

```diff
@@ -64,4 +66,102 @@ describe('Phase Allocation Endpoints', () => {
     const res = await getAllSitePhases(site.id);
     expect(res.length).not.toEqual(0);
   });
+
+  it('getAllSitePhases, returns correct # of hires and remaining hires per phase', async () => {
+    const numAllocations = 30;
+    const phaseMockYearOne = {
+      name: '2015 Phase',
+      start_date: '2015/01/01',
+      end_date: '2015/12/31',
+    };
+    const phaseMockYearTwo = {
+      name: '2016 Phase',
+      start_date: '2016/01/01',
+      end_date: '2016/12/31',
+    };
+    const phaseYearOne = await createPhase(phaseMockYearOne, user);
+    const phaseYearTwo = await createPhase(phaseMockYearTwo, user);
+
+    const siteMock = siteData({
+      siteId: 7,
+      siteName: 'Test phase',
+      operatorEmail: 'test.e2e.phase@hcap.io',
+    });
+    const site = await makeTestSite(siteMock);
+
+    // An allocation for each created phase
+    await createAllocation({
+      site_id: site.id,
+      phase_id: phaseYearOne.id,
+      allocation: numAllocations,
+    }, user);
+    await createAllocation({
+      site_id: site.id,
+      phase_id: phaseYearTwo.id,
+      allocation: numAllocations,
+    }, user);
+
+    // Helper function for creating a participant with a hired status
+    const makeHiredParticipant = async ({
+      emailAddress,
+      employerId,
+      siteId,
+      hiredDate,
+      startDate,
+    }) => {
+      const participant = await makeTestParticipant({ emailAddress });
+      await makeTestParticipantStatus({
+        participantId: participant.id,
+        employerId,
+        status: participantStatus.HIRED,
+        current: true,
+        data: {
+          site: siteId,
+          hiredDate,
+          startDate,
+          nonHcapOpportunity: false,
+        }
+      });
+    }
+
+    const hiredStartDates = [
+      {
+        hiredDate: '2015/02/20',
+        startDate: '2015/03/01',
+      },
+      {
+        hiredDate: '2015/10/13',
+        startDate: '2016/01/15',
+      },
+      {
+        hiredDate: '2016/02/15',
+        startDate: '2016/03/15',
+      },
+      {
+        hiredDate: '2017/04/15',  // Should not be in a phase
+        startDate: '2017/05/15',
+      },
+    ];
+
+    for (let i = 0; i < hiredStartDates.length; i++) {
+      await makeHiredParticipant({
```

Comment:
Is there a reason this needs to be a for loop? Since the participants are not inter-dependent, a `map` and `Promise.all` would probably be much cleaner and faster.

Reply:
No reason, just rushed it. It's a good idea